crud.py

```diff
@@ -56,25 +56,13 @@ def get_exercises(
     user_id: int | None = None,
 ):
     exercises = select(models.Exercise)
-    # first, filter the exercises by complexity, category and title
-    if complexity:
-        exercises = exercises.filter(models.Exercise.complexity == complexity)
-    if categories:
-        for category in categories:
-            exercises = exercises.filter(models.Exercise.category.contains(category))
-    if title_like:
-        if case_sensitive:
-            exercises = exercises.filter(models.Exercise.title.like(f"%{title_like}%"))
-        else:
-            exercises = exercises.filter(models.Exercise.title.ilike(f"%{title_like}%"))
-    # then, sort the exercises
+    # sort the exercises
     if order_by:
         # sort by completion (completion is in DoExercise table)
         if order_by == schemas.ExerciseOrderBy.completion:
-            exercises = (
-                select(models.Exercise)
-                .join(models.DoExercise)
-                .filter(models.DoExercise.user_id == user_id)
+            assert user_id
```

Comment:
We shouldn't leave asserts in the logic part of the application

Right now if we receive a request to sort by completion, and `user_id` is not provided (maybe a user mistake or something else), the whole app **crashes**

If we want to make sure that `user_id` is provided, or is not provided, I think it's better to do it in `main.py` as input validation. And if validation fails (id not provided for example) we should return an error, that specifies what went wrong `Can't sort by completion if not logged id`